python/tank/platform/engine.py

```diff
@@ -1219,6 +1212,53 @@ def _do_invoke(self):
         # don't have ui so can't create an invoker!
         return None
 
+    def _generate_panel_id(self, app_instance_name, panel_name):
```

Comment:
Is this broken out so that it can be subclassed or is it for modularity? If the latter, I think it should be `__prefixed`. If you are thinking of subclassing it in the engine, let's discuss that a bit!